lib/child_process.js

```diff
@@ -613,8 +596,37 @@ function normalizeSpawnArguments(file, args, options) {
 
 function spawn(file, args, options) {
   const child = new ChildProcess();
-
   options = normalizeSpawnArguments(file, args, options);
+
+  if (options.signal) {
+    const signal = options.signal;
+    // Validate signal, if present
+    validateAbortSignal(signal, 'options.signal');
+
+    // Do nothing and throw if already aborted
+    if (signal.aborted) {
+      onAbortListener();
+    } else {
+      const config = { once: true };
+      const remove = () => {
+        signal.removeEventListener('abort', onAbortListener, config);
+      };
+
+      signal.addEventListener('abort', onAbortListener, config);
```

Comment:
I don't think it's worth having a `config` variable if it's used only once.
```suggestion
      signal.addEventListener('abort', onAbortListener, { once: true });
```

Reply:
For this part, I saw that in this [comment](https://github.com/nodejs/node/pull/34249/#discussion_r559830442) you suggest me to remove it but here and also in the [documentation](https://nodejs.org/dist/latest-v15.x/docs/api/globals.html#globals_event_abort) said it's a valid arg. Should I keep it or remove it?

Reply:
I'm not suggesting removing `{ once: true }`, it's a valid arg for `addEventListener`: https://nodejs.org/dist/latest-v15.x/docs/api/events.html#events_eventtarget_addeventlistener_type_listener_options

However, there's no such argument for `removeEventListener`: https://nodejs.org/dist/latest-v15.x/docs/api/events.html#events_eventtarget_removeeventlistener_type_listener

Reply:
Oh, I missed that, thanks! Doing the change 👍

Reply:
Done in 026021a
Please have a look 🙂